spec/models/activerecord_obstacle_course_spec.rb

```diff
@@ -317,19 +319,17 @@
     unordered_items = [unordered_item_1, unordered_item_2, unordered_item_3]
 
     # ----------------------- Using Ruby -------------------------
-    items = Item.all
-
-    ordered_items = items.map do |item|
-      item if item.orders.present?
-    end.compact
-
-    ordered_items_names = ordered_items.map(&:name)
+    # items = Item.all
+    #
+    # ordered_items = items.map do |item|
+    #   item if item.orders.present?
+    # end.compact
+    #
+    # ordered_items_names = ordered_items.map(&:name)
     # ------------------------------------------------------------
-
     # ------------------ ActiveRecord Solution ----------------------
-    # Solution goes here
-    # When you find a solution, experiment with adjusting your method chaining
-    # Which ones are you able to switch around without relying on Ruby's Enumerable methods?
+    # ordered_items_names = Item.joins(:orders).uniq.pluck(:name)
+    ordered_items_names = Order.joins(:items).uniq.pluck(:name)
```

Comment:
I have a hard time making the claim that these are the *only two ways*, but they both look pretty good to me. 😄 

My gut tells me to prefer `distinct` to `uniq` as `distinct` is the AR method and `uniq` is a Ruby array method. Generally AR/SQL is going to be faster than Ruby. If this was something where you were really pushing for speed it's generally best to benchmark to confirm.